src/plan.rs

```diff
@@ -86,7 +91,7 @@ impl ExecutionPlan {
             self.global_ops.push(op);
         } else {
             for file_plan in &mut self.input_files {
-                file_plan.ops.push(op)
+                file_plan.ops.push(op.clone())
```

Comment:
My rust is not good enough (yet I hope) to find the best way out of this. `clone()` it is for now :D

Reply:
Cloning is the right call here.